packages/bot-analytics/src/db/token.ts

```diff
@@ -0,0 +1,53 @@
+import { typechain } from "@mangrovedao/mangrove.js";
+import { Token } from "@prisma/client";
+import { ChainContext } from "../types";
+import { PrismaTx } from "./types";
+
+export const generateCreateTokenIfNotExist = (context: ChainContext) => {
+  const ierc20 = typechain.IERC20__factory.createInterface();
+  return async (prisma: PrismaTx, address: string): Promise<Token> => {
+    let token = await prisma.token.findFirst({
+      where: {
+        address,
+        chainId: context.chainId,
+      },
+    });
+
+    if (token) {
+      return token;
+    }
+
+    const calls: typechain.Multicall2.CallStruct[] = [
+      {
+        target: address,
+        callData: ierc20.encodeFunctionData("symbol"),
+      },
+      {
+        target: address,
+        callData: ierc20.encodeFunctionData("decimals"),
+      },
+    ];
+
```

Comment:
I've been thinking about it since the beginning, I think it would be nice but it will require some archive node to run our subgraph. As we may run the graph ourself in the future, I preferred to do it here, but if you think it's a clear advantage then why not ?